src/server/hll_family.cc

```diff
@@ -211,13 +216,78 @@ void PFCount(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+OpResult<int> PFMergeInternal(CmdArgList args, ConnectionContext* cntx) {
+  try {
+    vector<vector<string>> hlls;
+    hlls.resize(shard_set->size());
+
+    atomic_bool success = true;
+    auto cb = [&](Transaction* t, EngineShard* shard) {
+      ShardId sid = shard->shard_id();
+      ArgSlice shard_args = t->GetShardArgs(shard->shard_id());
+      auto result = ReadValues(t->GetOpArgs(shard), shard_args);
+      if (result.ok()) {
+        hlls[sid] = std::move(result.value());
+      } else {
+        success = false;
+      }
+      return result.status();
+    };
+
+    Transaction* trans = cntx->transaction;
+    trans->Schedule();
+    trans->Execute(std::move(cb), false);
+
+    if (!success) {
+      return OpStatus::INVALID_VALUE;
```

Comment:
you can not exit prematuraly if you execute a non-final hop. You must send the final hop, I usually just send a noop that just returns OK.

Reply:
That's a good catch!
I wonder, why is this not causing a dead-lock in the `MergeInvalid` test? Or better, how can I change it to test that it doesn't?

Reply:
the deadlock manifests itself when you run more commands and then do not execute.
but for tests - there are multiple ways to test that we finished cleanly. we have `IsLocked` that checks per key.
Similarly, we can introduce `NumLocked()` that could return number of locked keys in all shards (we have similar NumWatched).
Finally, we can introduce `MaxTxLength()` that returns `min{txq().size()}` for all shards. Obviously it should be 0 when a test finishes.

Reply:
Added `NumLocked()`. I actually think that, perhaps, we should add this to all tests? (like in the test base class?)

Reply:
I added a global check, and 2 tests failed (see my changes).
If you want, I can look into why instead of blindly disabling the tests. But I imagine that it's not a good use of time, so I just disabled them. LMK.